tests/manage/cluster/nodes/test_nodes_restart.py

```diff
@@ -0,0 +1,206 @@
+import logging
+import pytest
+
+from ocs_ci.framework import config
+from ocs_ci.utility.utils import ceph_health_check
+from ocs_ci.ocs import ocp, constants
+from ocs_ci.framework.testlib import tier4, ignore_leftovers, ManageTest
+from ocs_ci.utility import aws
+from ocs_ci.ocs.cluster import CephCluster
+from tests import helpers
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def aws_obj():
+    """
+    Returns:
+        AWS: An instance of AWS class
+
+    """
+    aws_obj = aws.AWS()
+    return aws_obj
+
+
+@pytest.fixture()
+def instances(request, aws_obj):
+    """
+    Get cluster instances
+
+    Returns:
+        dict: The ID keys and the name values of the instances
+
+    """
+    nodes = ocp.get_all_nodes()
+    ec2_instances = aws.get_instances_ids_and_names(nodes)
+
+    def finalizer():
+        """
+        Make sure all instances are running
+        """
+        stopping_instances = {
+            key: val for key, val in ec2_instances.items() if (
+                aws_obj.get_instances_status_by_id(key) == constants.INSTANCE_STOPPING
+            )
+        }
+        if stopping_instances:
+            for stopping_instance in stopping_instances:
+                instance = aws_obj.get_ec2_instance(stopping_instance.key())
+                instance.wait_until_stopped()
+        stopped_instances = {
+            key: val for key, val in ec2_instances.items() if (
+                aws_obj.get_instances_status_by_id(key) == constants.INSTANCE_STOPPED
+            )
+        }
+        if stopped_instances:
+            aws_obj.start_ec2_instances(instances=stopped_instances, wait=True)
+
+    request.addfinalizer(finalizer)
+
+    return ec2_instances
+
+
+@pytest.fixture()
+def resources(request):
+    """
+    Delete the resources created during the test
+
+    Returns:
+        tuple: empty lists of resources
+
+    """
+    projects, secrets, pools, storageclasses, pvcs, pods = ([] for i in range(6))
```

Comment:
I suspect the uninitialized list might cause issues - maybe it'll be safer to initialize it using
```suggestion
    projects, secrets, pools, storageclasses, pvcs, pods = ([None] for i in range(6))
```